client/src/components/Navbar/Navbar.tsx

```diff
@@ -1,15 +1,69 @@
-import { Avatar, Box, Button, Grid } from '@material-ui/core';
+import React, { useState } from 'react';
+import { Avatar, Box, Button, Dialog, FormControl, Grid, IconButton, TextField, Typography } from '@material-ui/core';
 import useStyles from './useStyles';
 import logoImage from './logo.jpg';
 import { NavLink, BrowserRouter as Router } from 'react-router-dom';
 import DashboardIcon from '@material-ui/icons/Dashboard';
 import CalendarTodayIcon from '@material-ui/icons/CalendarToday';
+import Close from '@material-ui/icons/Close';
 import AddIcon from '@material-ui/icons/Add';
 import BoardsNavbar from '../BoardsNavbar/BoardsNavbar';
+import { DialogContent } from './useStyles';
+import MuiDialogTitle from '@material-ui/core/DialogTitle';
+
+export interface Props {
+  id: string;
+  children: React.ReactNode;
+  onClose: () => void;
+}
+export interface IBoards {
+  board: string;
+}
 
 const Navbar = (): JSX.Element => {
   const classes = useStyles();
-
+  const [open, setOpen] = React.useState(false);
+  const [title, setTitle] = useState('');
+  const [boards, setBoards] = useState<IBoards[]>([]);
+  const totalTitle = boards;
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const DialogTitleIcon = (props: Props) => {
+    const { onClose, ...other } = props;
+    return (
+      <MuiDialogTitle className={classes.iconTitleBox} disableTypography {...other}>
+        <Typography variant="h5" className={classes.cardTitle}>
+          {onClose ? (
+            <IconButton aria-label="close" onClick={onClose}>
+              <Close />
+            </IconButton>
+          ) : null}
```

Comment:
I personally don't like returning a `null` content. Instead I would put this code like this:
`
{!!onClose && (
            <IconButton aria-label="close" onClick={onClose}>
              <Close />
            </IconButton>
          ) }
`
So it renders only when `onClose` (converted to boolean with `!!`) is `true`
This is a personal preference, though :smiley: